sklearn/model_selection/_classification_threshold.py

```diff
@@ -686,37 +661,17 @@ class TunedThresholdClassifierCV(BaseThresholdClassifier):
         The classifier, fitted or not, for which we want to optimize
         the decision threshold used during `predict`.
 
-    objective_metric : {"max_tpr_at_tnr_constraint", "max_tnr_at_tpr_constraint", \
-            "max_precision_at_recall_constraint, "max_recall_at_precision_constraint"} \
-            , str, dict or callable, default="balanced_accuracy"
+    objective_metric : str, dict or callable, default="balanced_accuracy"
```

Comment:
shouldn't this be called `scoring` the same as `*SeachCV`? We're optimizing using a cv object optimizing for one or more metrics, exactly like `*SearchCV`, would be nice to have a consistent name for this.

Reply:
I agree that it makes the API more consistent

Reply:
I was trying to find where we got the discussion in the past but I could not find it.
I think that the fact that we had constraint metric was different enough from the "scoring" parameter that we decided to change the name.

Reply:
I'm pretty optimistic that we can find an API which we can share with `*SearchCV` which is both user friendly and capable of handling usecases we want. And as @jeremiedbb says in https://github.com/scikit-learn/scikit-learn/pull/26120#pullrequestreview-2038175696 , we're not far from nicely supporting it. So I'd say we can safely rename the parameter.

Reply:
Now I have a question: do we need `pos_label` and `response_method` parameters. Those two can be delegated to `make_scorer`: `response_method` is a parameter of this function while `pos_label` is a parameter of the function provided to `make_scorer` and forwarded to it.

Reply:
For instance, not exposing those is the current API of the `*SearchCV` that does not expose the `response_method` and `pos_label`. The downside is coming to "how many people are using `make_scorer`".

Reply:
But they will be requested for the `FixedThresholdClassifier` because we don't have any metric.

Reply:
I'm very much liking how this is simplifying the top level API of the estimator.

Reply:
> The downside is coming to "how many people are using make_scorer

the people who don't are fine with the defaults

Reply:
Currently, if I want to tune a LogisticRegression with a grid search and recall as scoring, either I pass ``scoring="recall"`` and rely on the defaults, or I pass a scorer made out of ``recall_score`` if I want to change the pos lable. I think it's fine to mimic this behavior

Reply:
> But they will be requested for the FixedThresholdClassifier because we don't have any metric.

For this one we need to keep them indeed I think because if a user passes a threshold of say 0.3, we need to know if it's for a proba or for a decision function.

Reply:
Actually, the response_method of the estimator and the response_method for the scorer are not the same, so we need to keep it.

Reply:
Indeed. `response_method` is dispatched to `_CurveScorer` and not the scorer. It would be weird to request our user passing `make_scorer(balanced_accuracy_score, response_method="predict_proba")` while the scorer alone does not make any sense.